### PR TITLE
system job labels

### PR DESCRIPTION
**Pull request by mmjconolly**
This adds some `Plan` `Label`s to some of the `Job`s in `share/wake/lib/system`.
It's somewhat useful when making manual SQL queries on `wake.db`

### CODE REVIEW

File: share/wake/lib/system/io.wake [hunk 1]
```diff
@@ -68,6 +69,7 @@ export def installAs (dest: String) (file: Path): Result Path Error =
         sdest, Nil
     makeExecPlan cmd inputs
     | setPlanEnvironment Nil
+    | setPlanLabel "installAs: {dest}"
```
Comment (ag-eitilt) [inline share/wake/lib/system/io.wake:72]: This one might want to record the source file as well.  Also, for `targz` and `claimFileAsPathIn`, you've used slightly more readable "job names", and I'd expect this to be `install-as:` to follow their pattern.

File: share/wake/lib/system/path.wake [hunk 2]
```diff
@@ -124,7 +124,7 @@ target hashcode (f: String): String =
   if reuse !=* "" then reuse else
     def hashPlan cmd  =
       Plan "" cmd Nil Nil "." "" logNever logError logDebug ReRun True Nil (\_ True) hashUsage id id
-    def job = hashPlan ("<hash>", f, Nil) | runJobWith localRunner
+    def job = hashPlan ("<hash>", f, Nil) | setPlanLabel "hash: {f}" | runJobWith localRunner
```
Comment (ag-eitilt) [inline share/wake/lib/system/path.wake:127]: I don't know the full logic; will these start showing up in logs/stdout (maybe with a `-v`, or a `--no-tty`)?  The others are relevant enough to the user that I wouldn't mind having them be shown -- though we want to be careful to not wind up with huge CI logs -- but hashing in particular seems like an internal job that we would kind of prefer to keep hidden.

File: share/wake/lib/system/sources.wake [hunk 3]
```diff
@@ -155,5 +157,6 @@ export def claimFileAsPathIn (outputDirectory: Path) (existingFile: String) (des
         | setPlanLocalOnly   True
         | setPlanPersistence Once
         | setPlanFnOutputs   (\_ desiredWorkspacePath, Nil)
+        | setPlanLabel "claim-in: {existingFile}"
```
Comment (ag-eitilt) [inline share/wake/lib/system/sources.wake:160]: As a mirror, this might want to record the destination.

**Review by ag-eitilt — APPROVED** [review]
Looks good, conditional on the response to the second comment.

**Review by terpstra — COMMENTED** [review]
test

**Review by terpstra — APPROVED** [review]

**Comment by JakeSiFive** [comment]
Can we make this use Ashley's new `@!` feature and land this? I think this is a good change.

**Comment by mmjconolly** [comment]
What does `@!` do?

**Comment by ag-eitilt** [comment]
It's like `@here`, adding the line number: `@! ==~ "wake/build-graph.wake:226"`.

**Pull request closed without merging**